src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -388,12 +396,20 @@ void IList.Remove(object value)
 
         void ICollection.CopyTo(Array array, int index)
         {
-            if (array == null)
-                throw new ArgumentNullException(nameof(array));
-            if (array.Rank != 1)
-                throw new ArgumentException(nameof(array));
+            if ((array != null) && (array.Rank != 1))
+            {
+                throw new ArgumentException(); // Arg_RankMultiDimNotSupported
+            }
 
-            Array.Copy(_items, 0, array, index, _size);
+            try
+            {
+                // Array.Copy will check for null.
```

Comment:
@AlexRadch, `List` and `ReadOnlyCollectionBuilder` are two separate classes (and subclassing one from the latter would be a breaking change, in response to your earlier question). I think it is best to keep the changes consistent with what is already there rather than what is in `List`. It doesn't seem valuable to make this change.